substratevm/src/com.oracle.objectfile/src/com/oracle/objectfile/pecoff/cv/CVRootPackages.java

```diff
@@ -0,0 +1,217 @@
+                    "org.graalvm.compiler.truffle.runtime",
+                    "org.graalvm.compiler.truffle.runtime.hotspot",
+                    "org.graalvm.compiler.truffle.runtime.hotspot.java",
+                    "org.graalvm.compiler.truffle.runtime.hotspot.jdk8+13",
+                    "org.graalvm.compiler.truffle.runtime.hotspot.libgraal",
+                    "org.graalvm.compiler.truffle.runtime.serviceprovider",
+                    "org.graalvm.compiler.truffle.runtime.serviceprovider.jdk8",
+                    "org.graalvm.compiler.virtual",
+                    "org.graalvm.compiler.virtual.bench",
+                    "org.graalvm.compiler.word",
+                    "org.graalvm.graphio",
+                    "org.graalvm.libgraal",
+                    "org.graalvm.libgraal.jdk8",
+                    "org.graalvm.micro.benchmarks",
+                    "org.graalvm.util",
+    };
+
+    private static final String[] intrinsicClassNames = {
+                    "com.oracle.svm.core.genscavenge.AlignedHeapChunk",
+                    "com.oracle.svm.core.genscavenge.CardTable",
+                    "com.oracle.svm.core.genscavenge.ObjectHeaderImpl",
+                    "com.oracle.svm.core.genscavenge.graal.GenScavengeAllocationSnippets",
+                    "com.oracle.svm.core.genscavenge.graal.BarrierSnippets",
+                    "com.oracle.svm.core.snippets.KnownIntrinsics",
+                    "com.oracle.svm.core.graal.snippets.SubstrateAllocationSnippets",
+                    "com.oracle.svm.core.threadlocal.FastThreadLocalBytes",
+                    "org.graalvm.compiler.replacements.AllocationSnippets",
+                    "org.graalvm.compiler.nodes.PrefetchAllocateNode",
+                    "com.oracle.svm.core.os.CopyingImageHeapProvider"
+    };
+
+    private static final HashSet<String> rootPackageSet;
+    private static final HashSet<String> intrinsicClassNameSet;
+
+    static {
+        rootPackageSet = new HashSet<>(rootPackageNames.length);
+        Collections.addAll(rootPackageSet, rootPackageNames);
+        intrinsicClassNameSet = new HashSet<>(intrinsicClassNames.length);
+        Collections.addAll(intrinsicClassNameSet, intrinsicClassNames);
+    }
+
+    static boolean isGraalPackage(String pn) {
+        return rootPackageSet.contains(pn);
+    }
+
+    private static String getPackagename(String className) {
+        return className.contains(".") ? className.substring(0, className.lastIndexOf('.')) : className;
+    }
+
+    static boolean isGraalClass(String cn) {
+        final String pn = getPackagename(cn);
+        return isGraalPackage(pn);
+    }
+
+    /**
+     * is class a Graal intrinsic class?
+     *
+     * @param cn class name of code
+     * @return true if this is Graal intrinsic code
+     */
+    static boolean isGraalIntrinsic(String cn) {
+        return intrinsicClassNameSet.contains(cn);
+    }
+
+    static boolean isJavaPackage(String pn) {
```

Comment:
I think there may be a problem here that currently exists in the SourceCache code. As far as the source caches are concerned the package filter on "javax.* packages" is used to identify that a class belongs to the JDK runtime and that it's source can be found in the src.zip distributed with the JDK. I don't think that is strictly correct as I believe there are implementations of javax.* packages in Java EE components i.e. which ought to be treated as application classes. I'm not clear but I suspect the same problem applies with your test here. Anyway, I'm also still dubious as to why users might want to be able not to see JDK runtime code given the assumed use case for this debug capability.